scripts/uploadWhitelist.js

```diff
@@ -0,0 +1,177 @@
+/* eslint-disable no-console */
+
+// eslint-disable-next-line import/no-extraneous-dependencies
+require("babel-register");
+const d3 = require("d3-dsv");
+const fs = require("fs");
+const confirm = require("node-ask").confirm;
+const path = require("path");
+
+const Commitment = artifacts.require("Commitment");
+
+const tokenEnum = {
+  EUR: 2,
+  ETH: 1
+};
+
+const Q18 = new web3.BigNumber(10).pow(18);
+
+const isAddress = address => {
+  if (!web3.isAddress(address))
+    throw new Error(
+      `Investor with Address:${address} has wrong address format!!`
+    );
+  return address;
+};
+const isCurrency = (currency, address) => {
+  if (currency !== "EUR" && currency !== "ETH")
+    throw new Error(
+      `Investor:${address} and currency:${currency} has wrong currency format!!`
+    );
+  return tokenEnum[currency];
+};
+const getAmount = (amount, address) => {
+  const investAmount = Number.parseFloat(amount.replace(/^\D+/g, ""));
+  // Most of the case this is for an empty string
+  if (Number.isNaN(investAmount))
+    throw new Error(`Investor ${address} has their amount left out`);
+  return investAmount === 0 ? new web3.BigNumber(0) : Q18.mul(investAmount);
+};
+
+const getAttributes = (address, currency, amount) => ({
+  wlAddresses: isAddress(address),
+  wlTokens: isCurrency(currency, address),
+  wlTickets: getAmount(amount, address)
+});
+const isDuplicate = array => {
+  const duplicateArray = array.filter(
+    data =>
+      array.filter(investor => {
+        if (investor.wlAddresses === data.wlAddresses) return true;
+        return undefined;
+      }).length > 1
+  );
+  if (duplicateArray.length > 0) {
+    console.log("You your duplicate object");
+    console.log(duplicateArray);
+    throw new Error("You have a duplications");
+  }
+};
+const filterWlfromSmartContract = async (filteredWhiteList, commitment) => {
+  let lastUploadedfile = false;
+  const verifiedWhiteList = (await Promise.all(
+    filteredWhiteList.map(async investor => {
+      const whiteListTicketbyAddress = await commitment.whitelistTicket(
+        investor.wlAddresses
+      );
+      const tokenType = whiteListTicketbyAddress[0].isZero();
+      const ticketSize = whiteListTicketbyAddress[1].toString();
+      const ticketSizefromList =
+        investor.wlTokens === 1
+          ? (await commitment.convertToEur(investor.wlTickets)).toString()
+          : investor.wlTickets.toString();
+
+      if (ticketSize !== ticketSizefromList && tokenType === false) {
```

Comment:
`tokenType === false` what does it mean here. if tokenType === 0 we should immediately show that expected investor is not on the list